### PR TITLE
Add organization claim to backend JWT

### PR DESCRIPTION
**Pull request by fazlan-nazeem**
addresses wso2-enterprise/choreo/issues/8470

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/utils/GatewayUtils.java [hunk 1]
```diff
@@ -981,16 +981,20 @@ public static JWTInfoDto generateJWTInfoDto(JWTValidationInfo jwtValidationInfo,
...
-            if (jwtInfoDto.getJwtValidationInfo().getClaims() != null
-                    && jwtInfoDto.getJwtValidationInfo().getClaims().get("sub") != null) {
-                String sub = (String) jwtInfoDto.getJwtValidationInfo().getClaims().get("sub");
-                jwtInfoDto.setSub(MultitenantUtils.getTenantAwareUsername(sub));
+        if (jwtInfoDto.getJwtValidationInfo().getClaims() != null) {
+            Map<String, Object> claims = jwtInfoDto.getJwtValidationInfo().getClaims();
+            String subClaim = "sub";
+            String organizationsClaim = "organizations";
+            if (claims.get(subClaim) != null) {
+                String sub = (String) jwtInfoDto.getJwtValidationInfo().getClaims().get(subClaim);
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/utils/GatewayUtils.java:989]:
```suggestion
                String sub = (String) claims.get(subClaim);
```

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/utils/GatewayUtils.java [hunk 2]
```diff
@@ -981,16 +981,20 @@ public static JWTInfoDto generateJWTInfoDto(JWTValidationInfo jwtValidationInfo,
...
+                jwtInfoDto.setSub(sub);
+            }
+            if (claims.get(organizationsClaim) != null) {
+                String[] organizations = (String[]) jwtInfoDto.getJwtValidationInfo().getClaims().
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/utils/GatewayUtils.java:993]:
```suggestion
                String[] organizations = (String[]) claims.
```

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/utils/GatewayUtils.java [hunk 3]
```diff
@@ -981,16 +981,20 @@ public static JWTInfoDto generateJWTInfoDto(JWTValidationInfo jwtValidationInfo,
...
+                        get(organizationsClaim);
+                jwtInfoDto.setOrganizations(organizations);
             }
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/utils/GatewayUtils.java:996]:
```suggestion
            String sub = (String)claims.get(subClaim);
            String[] organizations = (String[])claims.get(organizationsClaim);
            if (StringUtils.isNotBlank(sub)) {
                jwtInfoDto.setSub(sub);
            }
            if (organizations != null) {
                jwtInfoDto.setOrganizations(organizations);
            }
```

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/utils/GatewayUtils.java [hunk 4]
```diff
@@ -981,16 +981,20 @@ public static JWTInfoDto generateJWTInfoDto(JWTValidationInfo jwtValidationInfo,
...
                                             APIKeyValidationInfoDTO apiKeyValidationInfoDTO, JWTInfoDto jwtInfoDto) {
 
-        if (jwtInfoDto.getJwtValidationInfo() != null) {
-            jwtInfoDto.setEndUser(getEndUserFromJWTValidationInfo(jwtInfoDto.getJwtValidationInfo(),
-                    apiKeyValidationInfoDTO));
-            if (jwtInfoDto.getJwtValidationInfo().getClaims() != null
-                    && jwtInfoDto.getJwtValidationInfo().getClaims().get("sub") != null) {
-                String sub = (String) jwtInfoDto.getJwtValidationInfo().getClaims().get("sub");
-                jwtInfoDto.setSub(MultitenantUtils.getTenantAwareUsername(sub));
+        if (jwtInfoDto.getJwtValidationInfo().getClaims() != null) {
```
[the remark this answers is not in the document]
Comment (fazlan-nazeem) [reply components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/utils/GatewayUtils.java]: Fixed with 05e7f2c4f74f9b1beb7ad879db49c96dbf688719

**Pull request merged**